tests/integration/thrift_tests.py

```diff
@@ -72,13 +72,16 @@ def raw_thrift_client(endpoint, client_spec):
 
 
 @contextlib.contextmanager
-def baseplate_thrift_client(endpoint, client_spec, client_span_observer=None):
-    baseplate = Baseplate(
-        app_config={
-            "baseplate.service_name": "fancy test client",
-            "example_service.endpoint": str(endpoint),
-        }
-    )
+def baseplate_thrift_client(
+    endpoint, client_spec, client_span_observer=None, budget_header_test=False,
+):
+    app_config = {
+        "baseplate.service_name": "fancy test client",
+        "example_service.endpoint": str(endpoint),
+    }
+    if budget_header_test:
+        app_config["example_service.timeout"] = "1000 seconds"
```

Comment:
it's probably better to also pass `1000 seconds` part in as an arg of `baseplate_thrift_client`.

e.g. make the arg actually `timeout=None`, then here we can do

```python
if timeout
  app_config["example_service.timeout"] = timeout
```

instead.